src/WebJobs.Script/Config/ConsoleLoggingOptionsSetup.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    public class ConsoleLoggingOptionsSetup : IConfigureOptions<ConsoleLoggingOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public ConsoleLoggingOptionsSetup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(ConsoleLoggingOptions options)
+        {
+            options.LoggingDisabled = _configuration.GetValue<int>(EnvironmentSettingNames.ConsoleLoggingDisabled) == 1;
+
+            int bufferSize = _configuration.GetValue<int>(EnvironmentSettingNames.ConsoleLoggingBufferSize, ConsoleLoggingOptions.DefaultBufferSize);
+            if (bufferSize < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(EnvironmentSettingNames.ConsoleLoggingBufferSize), "Console buffer size cannot be negative");
+            }
+
+            options.BufferEnabled = bufferSize > 0;
```

Comment:
This feels a bit unintuitive to me: we will ignore any explicit `BufferEnabled = false` settings, and always set it based on `BufferSize`. Two ways I think this can be fixed:

1. Make `BufferEnabled` `internal`, so only `BufferSize` is exposed publicly and document that setting to `0` disables buffer.
2. Make `BufferEnabled` a `bool?` and only set it if it is still `null`.

Reply:
OK, making buffer size setter internal.